ra/ra.go

```diff
@@ -283,13 +283,16 @@ func (ra *RegistrationAuthorityImpl) NewRegistration(ctx context.Context, init c
 	return
 }
 
-func (ra *RegistrationAuthorityImpl) validateContacts(ctx context.Context, contacts []*core.AcmeURL) error {
-	if ra.maxContactsPerReg > 0 && len(contacts) > ra.maxContactsPerReg {
+func (ra *RegistrationAuthorityImpl) validateContacts(ctx context.Context, contacts *[]*core.AcmeURL) error {
+	if contacts == nil {
```

Comment:
For clarity let's make this `if contacts == nil || len(*contacts) == 0`. Otherwise the two different "nothing to validate" cases follow different paths through the function.